packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.js

```diff
@@ -14,50 +14,118 @@ import noMatchesFor from '@salesforce/label/c.quantic_NoMatchesFor';
 import collapseFacet from '@salesforce/label/c.quantic_CollapseFacet';
 import expandFacet from '@salesforce/label/c.quantic_ExpandFacet';
 
+/** @typedef {import("coveo").SearchEngine} SearchEngine */
+/** @typedef {import("coveo").CategoryFacet} CategoryFacet */
+/** @typedef {import("coveo").CategoryFacetState} CategoryFacetState */
+/** @typedef {import("coveo").CategoryFacetValue} CategoryFacetValue */
+
+/**
+ * A facet is a list of values for a certain field occurring in the results, ordered using a configurable criterion (e.g., number of occurrences).
+ * A `QuanticCategoryFacet` displays field values in a browsable, hierarchical fashion.
+ * @example
+ * <c-quantic-category-facet engine-id={engineId} facet-id="myfacet" field="geographicalhierarchy" label="Country" base-path="Africa,Togo,Lome" no-filter-by-base-path delimiting-character="/" number-of-values="5" is-collapsed></c-quantic-category-facet>
+ */
 export default class QuanticCategoryFacet extends LightningElement {
-  /** @type {import("coveo").CategoryFacetState} */
-  // @ts-ignore TODO: Check CategoryFacetState typing and integration with LWC/Quantic
-  @track state = {
-    values: [],
-    parents: [],
-  };
-
-  /** @type {string} */
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /** 
+   * A unique identifier for the facet.
+   * Defaults to the `field` value.
+   * @api
+   * @type {string}
+   * @defaultValue Defaults to the `field` value.
+   */
   @api facetId;
-  /** @type {string} */
+  /**
+   * The field whose values you want to display in the facet.
+   * @api
+   * @type {string}
+   */
   @api field;
-  /** @type {string} */
+  /**
+   * The non-localized label for the facet. This label is displayed in the facet header.
+   * @api
+   * @type {string}
+   */
   @api label = 'no-label';
-  /** @type {string} */
-  @api engineId;
-  /** @type {boolean} */
-  @api withSearch = false;
-  /** @type {string} */
+  /**
+   * The base path shared by all facet values, separated by commas.
+   * @api
+   * @type {string}
+   * @defaultValue `''`
+   */
   @api basePath = '';
-  /** @type {boolean} */
+  /**
+   * Whether not to use the `basePath` as a filter for the results.
+   * @api
+   * @type {boolean}
+   * @defaultValue `false`
+   */
   @api noFilterByBasePath = false;
-  /** @type {boolean} */
+  /**
+   * Whether not to exclude the parents of folded results when estimating the result count for each facet value.
+   * @api
+   * @type {boolean}
+   * @defaultValue `false
+   */  
   @api noFilterFacetCount = false;
-  /** @type {string} */
+  /**
+   * The character that separates the values of the target multi-value field.
```

Comment:
It would be nice to indicate that `delimitingCharacter` represents the "level delimiter" of the path. As discussed offline, it differs depending on how the field has been defined in Coveo Cloud.

If the field is defined as "hierarchical", parts of a path are delimited by `;`. A value is indexed as `parent;child` and `delimitingCharacter` should be set to `;`.

If the field is defined as a "multi-value", each path is delimited by `;`, and each part of a path is delimited by `>`. So the indexed value looks like `parent1>child1; parent2>child2`, and `delimitingCharacter` should be set to `>`.